src/main/java/fr/insalyon/pldagile/view/maps/MapPin.java

```diff
@@ -0,0 +1,34 @@
+package fr.insalyon.pldagile.view.maps;
+
+import fr.insalyon.pldagile.model.RequestType;
+import javafx.scene.image.ImageView;
+
+public abstract class MapPin extends ImageView {
+    static protected double DEFAULT_ICON_SIZE = 35D;
+    protected double iconSize = DEFAULT_ICON_SIZE;
+
+    /**
+     * Adjusts the coordinates of the view so that their origin is at the bottom center of
+     * the pin and not at its top left corner
+     */
+    protected void centerCoordinates(){
+        setCenteredX();
+        setCenteredY();
+    }
+
+    /**
+     * Sets the new Y origin on the icon
+     */
+    private void setCenteredY(){
+        setY(-iconSize);
+    }
+
+    /**
+     * Sets the new X origin on the icon
+     */
+    private void setCenteredX(){
+        super.setX(-iconSize / 2.6); // Somehow, dividing by 2 makes it not on the intersection
```

Comment:
set a variable that has a value of 2.6, rather than just putting a value inside setX.

Reply:
hmm k

Reply:
I didn't do that because i wanted to add a comment explaining why I'm dividing by 2.6 and not 2 (which would be taking half of the width). Plus in that case idk why we would store this value, making the comment for it inaccessible seems like a bigger advantage than the advantage of making the code look a bit cleaner